meson.py

```diff
@@ -14,13 +14,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from mesonbuild import mesonmain
-import sys, os
+import sys
+from pathlib import Path
+
+# If we're run uninstalled, add the script directory to sys.path to ensure that
+# we always import the correct mesonbuild modules even if PYTHONPATH is mangled
+meson_exe = Path(sys.argv[0]).resolve()
+if (meson_exe.parent / 'mesonbuild').is_dir():
+    sys.path.insert(0, meson_exe.parent)
```

Comment:
`meson_exe.parent` has type `pathlib.Path`, you should call `str()` on it.

Reply:
You are correct, Python 3.5 does not automatically convert that to str. Fixing.

How did you find this? Did you see a failure because of it?

Reply:
I was playing with `mypy`.

Reply:
If you get that working well, please open a PR to add it to the CI :)